src/main/java/com/google/devtools/build/lib/analysis/test/TestActionBuilder.java

```diff
@@ -262,6 +272,26 @@ private TestParams createTestAction(int shards) {
         extraTestEnv.put(CC_CODE_COVERAGE_SCRIPT, collectCcCoverage.getExecPathString());
       }
 
+      if (!instrumentedFiles.getReportedToActualSources().isEmpty()) {
+        Artifact reportedToActualSourcesArtifact =
+            ruleContext.getUniqueDirectoryArtifact("_coverage_helpers","reported_to_actual_sources.txt");
+        String reportedToActualSourcesFileContent =
```

Comment:
This is roughly an equivalent of iterating a nested set in analysis, isn't it? Is there an easy way to move this to execution?